backend-go/lib.go

```diff
@@ -0,0 +1,229 @@
+// Package telestion provides a framework for building [Telestion] services in Go.
+//
+// [Telestion]: https://telestion.wuespace.de/
+package telestion
+
+import (
+	"errors"
+	"github.com/nats-io/nats.go"
+	"path/filepath"
+)
+
+// Config configures the Telestion service.
+// The parsed configuration returned from the startService function for further usage.
+type Config struct {
+
+	// Dev adds minimal required configuration parameters to the configuration object.
+	Dev bool `mapstructure:"DEV"`
+
+	// NatsUrl describes the url that NATS uses to connect to the NATS server.
+	NatsUrl string `mapstructure:"NATS_URL"`
+
+	// NatsUser is the username that the NATS client uses to log in on the NATS server.
+	// This property can be undefined if no configuration source provided this parameter.
+	NatsUser string `mapstructure:"NATS_USER"`
+
+	// NatsPassword is the password that the NATS client uses to log in on the NATS server.
+	// This property can be undefined if no configuration source provided this parameter.
+	NatsPassword string `mapstructure:"NATS_PASSWORD"`
+
+	// ConfigFile contains the path to the configuration file that [StartService] read during startup.
+	// This property can be undefined if no configuration source provided this parameter.
+	ConfigFile string `mapstructure:"CONFIG_FILE"`
+
+	// ConfigKey describes a key in config file's root object that configures this service.
+	// This property can be undefined if no configuration source provided this parameter.
+	ConfigKey string `mapstructure:"CONFIG_KEY"`
+
+	// ServiceName contains the name of the service.
+	// This is used to create a unique queue group for the service.
+	ServiceName string `mapstructure:"SERVICE_NAME"`
```

Comment:
Describe the use case for queue groups:

```suggestion
	// ServiceName contains the name of the service.
	// This is used to create a unique queue group for the service and support running multiple 
    // parallel instances.
	ServiceName string `mapstructure:"SERVICE_NAME"`
```